cub/cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -844,6 +845,12 @@ public:
   // Pass-through bin transform operator
   struct PassThruTransform
   {
+// GCC 14 rightfully warns that when a value-initialized array of this struct is copied using memcpy, uninitialized
+// bytes may be accessed. To avoid this, we add a dummy member, so value initialization actually initializes the memory.
+#if defined(_CCCL_COMPILER_GCC) && __GNUC__ == 14
+    char dummy;
+#endif
```

Comment:
Question: Would it suffice to provide a user defined constructor that does nothing? That should technically zero out struct

Reply:
I can try!